multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/ProcessConflictPreventer.java

```diff
@@ -45,17 +46,27 @@ public synchronized void acquireLock(String mtaId, String namespace, String spac
         LOGGER.info(format(Messages.ACQUIRED_LOCK, processId, StepsUtil.getQualifiedMtaId(mtaId, namespace)));
     }
 
-    private void validateNoConflictingOperationsExist(String mtaId, String namespace, String spaceId) {
+    private void validateNoConflictingOperationsExist(String mtaId, String namespace, String spaceId, String processId) {
         List<Operation> conflictingOperations = findConflictingOperations(mtaId, namespace, spaceId);
         String qualifiedMtaId = StepsUtil.getQualifiedMtaId(mtaId, namespace);
 
-        if (conflictingOperations.size() == 1) {
+        if (conflictingOperations.stream()
+                                 .anyMatch(Operation::hasAcquiredLock)) {
+            if (conflictingOperations.stream()
+                                     .filter(Operation::hasAcquiredLock)
+                                     .count() >= 2) {
+                List<String> operationIds = getOperationIds(conflictingOperations);
+                throw new SLException(Messages.MULTIPLE_OPERATIONS_WITH_LOCK_FOUND, qualifiedMtaId, spaceId, operationIds);
+            }
             Operation conflictingOperation = conflictingOperations.get(0);
             throw new SLException(Messages.CONFLICTING_PROCESS_FOUND, conflictingOperation.getProcessId(), qualifiedMtaId);
         }
-        if (conflictingOperations.size() >= 2) {
-            List<String> operationIds = getOperationIds(conflictingOperations);
-            throw new SLException(Messages.MULTIPLE_OPERATIONS_WITH_LOCK_FOUND, qualifiedMtaId, spaceId, operationIds);
+
+        if (!conflictingOperations.isEmpty()) {
+            var latestOperation = conflictingOperations.get(0);
```

Comment:
Did you consider/make test with SLP scenario where you can trigger multiple mta deployments with single process?

Reply:
Would they be concurrent for the same MTA?